tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -105,10 +105,16 @@ dwarf_init(const char *filepath,
 	/* Populate the built-in types */
 	ret = populateBuiltInTypeDies(error);
 
+	/* Get the path of the file descriptor */
+	buffer = strdup("/proc/");
```

Comment:
This code actually can segfault as you have written in.

1. strdup("/proc/") will malloc a char array that can hold precisely 6 characters.
2. Then toString(getpid()) creates a string object, allocating more memory.
3. Now, you have a char [7] == {"/proc/"} and you are concatenating the string object converted to a char array to the end of it, overrunning the end of the buffer.

Instead of malloc'ing a tiny array on the heap with strdup() and creating 2 std::string objects then freeing it, why not just create a local array of an appropriate size and put it all together with one sprintf()? General coding practice is to not allocate heap memory for something (buffer) you are only using temporarily/locally.

This will use a local variable instead of the heap, not overrun your buffer, and not wastefully create and convert std::strings:
char buffer[100] = {'/0'};
sprintf(buffer, "/proc/%d/fd/%d", getpid(), fd);